src/Vitest.res

```diff
@@ -1180,29 +1194,121 @@ module Vi = {
+   */
+  type fakeTimersConfig = {
+    now?: Js.Date.t, // or int
+    toFake?: array<string>,
+    loopLimit?: int,
+    shouldAdvanceTime?: bool,
+    advanceTimeDelta?: int,
+    shouldClearNativeTimers?: bool,
+  }
+
+  @send external useFakeTimers: (t, ~config: fakeTimersConfig=?, unit) => t = "useFakeTimers"
+  @inline let useFakeTimers = (~config=?, ()) => vi_obj->useFakeTimers(~config?, ())
 
   @send external useRealTimers: t => t = "useRealTimers"
   @inline let useRealTimers = () => vi_obj->useRealTimers
 
-  @send external mockCurrentDate: (t, Js.Date.t) => t = "mockCurrentDate"
-  @inline let mockCurrentDate = date => vi_obj->mockCurrentDate(date)
+  @send external isFakeTimers: t => bool = "isFakeTimers"
+  let isFakeTimers = () => vi_obj->isFakeTimers
+
+  @send @return(nullable)
+  external getMockedSystemTime: t => option<Js.Date.t> = "getMockedSystemTime"
+  let getMockedSystemTime = () => vi_obj->getMockedSystemTime
+
+  @send external getRealSystemTime: t => float = "getRealSystemTime"
+  let getRealSystemTime = () => vi_obj->getRealSystemTime
 
-  @send external restoreCurrentDate: (t, Js.Date.t) => t = "restoreCurrentDate"
-  @inline let restoreCurrentDate = date => vi_obj->restoreCurrentDate(date)
+  type waitForOptions = {
+    timeout?: int,
+    interval?: int,
+  }
+
+  @send external waitFor: (t, @uncurry unit => 'a, waitForOptions) => promise<'a> = "waitFor"
+
+  /**
+  @since(vitest >= 0.34.5)
+   */
+  let waitFor = (callback, ~timeout=?, ~interval=?, ()) => {
+    waitFor(vi_obj, callback, {?timeout, ?interval})
+  }
+
+  @send
+  external waitForAsync: (t, @uncurry unit => promise<'a>, waitForOptions) => promise<'a> = "waitFor"
+
+  /**
+  @since(vitest >= 0.34.5)
+   */
+  let waitForAsync = (callback, ~timeout=?, ~interval=?, ()) => {
+    waitForAsync(vi_obj, callback, {?timeout, ?interval})
+  }
+
+  type waitUntilOptions = {
+    timeout?: int,
+    interval?: int,
+  }
+
+  @send
+  external waitUntil: (t, @uncurry unit => 'a, waitUntilOptions) => promise<'a> = "waitUntil"
+
+  /**
+  @since(vitest >= 0.34.5)
+   */
+  let waitUntil = (callback, ~timeout=?, ~interval=?, ()) => {
+    waitUntil(vi_obj, callback, {?timeout, ?interval})
+  }
+
+  @send
+  external waitUntilAsync: (t, @uncurry unit => promise<'a>, waitUntilOptions) => promise<'a> =
+    "waitUntil"
+
+  /**
+  @since(vitest >= 0.34.5)
+   */
+  let waitUntilAsync = (callback, ~timeout=?, ~interval=?, ()) => {
+    waitUntilAsync(vi_obj, callback, {?timeout, ?interval})
+  }
 
-  @send external getMockedDate: t => Js.null<Js.Date.t> = "getMockedDate"
-  @inline let getMockedDate = () => vi_obj->getMockedDate->Js.Null.toOption
+  // binding this using vi_obj causes a runtime error. this is because vitest sees this inside this file, then tries to evaluate the hoisted function, but the hoisted function is not provided yet, it's just a parameter to the function
+  @send external hoisted: (t, @uncurry unit => 'a) => 'a = "hoisted"
```

Comment:
This was the cause of the CI error, vitest was trying to hoist this inside `Vitest.mjs` and it was causing a syntax error. So this can be either left like this or I can just leave the comment with the binding commented out